tests/foreman/ui/test_host.py

```diff
@@ -2387,3 +2387,52 @@ def test_positive_tracer_enable_reload(tracer_install_host, target_sat):
         )
         tracer = session.host_new.get_tracer(tracer_install_host.hostname)
         assert tracer['title'] == "No applications to restart"
+
+
+def test_positive_host_registration_with_non_admin_user(
+    test_name,
+    module_entitlement_manifest_org,
+    module_location,
+    target_sat,
+    rhel8_contenthost,
+    module_ak,
+):
+    """Register hosts from a non-admin user with only register_hosts, edit_hosts
+    and view_organization permissions
+
+    :id: 35458bbc-4556-41b9-ba26-ae0b15179731
+
+    :expectedresults: User with register hosts permission able to do it.
+
+    :CaseLevel: System
+    """
+
+    user_password = gen_string('alpha')
+    org = module_entitlement_manifest_org
+    role = target_sat.api.Role(organization=[org]).create()
+    target_sat.api_factory.create_role_permissions(
+        role, {'Organization': ['view_organizations'], 'Host': ['view_hosts', 'register_hosts']}
+    )
+    user = target_sat.api.User(
+        role=[role],
+        admin=False,
+        password=user_password,
+        organization=[org],
+        location=[module_location],
+        default_organization=org,
+        default_location=module_location,
+    ).create()
+    created_host = target_sat.api.Host(location=module_location, organization=org).create()
```

Comment:
why creating a separate host when you've rhel8_contenthost already, and you check this host['breadcrumb'] with it post-registration ?

Reply:
Looks like we are doing two different things here @shweta83.

Reply:
Yeah. I removed the other host.